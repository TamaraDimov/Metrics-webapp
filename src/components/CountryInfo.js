import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { DateTime } from 'luxon'; // corrected import statement
import { fetchCountries } from '../redux/countrySlice';

const CountryInfo = () => {
  const { countryId } = useParams();
  const dispatch = useDispatch();
  const countries = useSelector((state) => state.countries);

  useEffect(() => {
    if (countries.length === 0) {
      dispatch(fetchCountries());
    }
  }, [countries, dispatch]);

  const country = countries.find((c) => c.name === countryId);

  useEffect(() => {
    const clockElement = document.getElementById('clock'); // moved inside useEffect
    const runInterval = () => {
      setInterval(() => {
        const now = DateTime.now();
        clockElement.textContent = now.toLocaleString(
          DateTime.DATETIME_FULL_WITH_SECONDS,
        );
      }, 1000);
    };
    runInterval();
  }, []); // added empty dependency array

  return (
    <>
      {country && (
        <div className="countryInformation">
          <div className="modal">
            <img src={country.flagSvg} alt={country.flagAlt} className="flag" />
            <h1 className="name">{country.name}</h1>
          </div>
          <ul className="details">
            <li className="detail">
              <span className="modalName">Capital:</span>
              {' '}
              <span className="modalValue">{country.capital}</span>
            </li>
            <li className="detail">
              <span className="modalName">Population:</span>
              {' '}
              <span className="modalValue">
                {country.population.toLocaleString()}
              </span>
            </li>
            <li className="detail">
              <span className="modalName">Currencies:</span>
              {' '}
              <span className="modalValue">
                {Object.values(country.currencies).map((currency) => (
                  <span key={currency.symbol}>
                    {currency.symbol}
                    {' '}
                    -&nbsp;
                    {currency.name}
                  </span>
                ))}
              </span>
            </li>
            <li className="detail">
              <span className="modalName">Timezones:</span>
              {' '}
              <span className="modalValue">
                {country.timezones.map((timezone) => (
                  <span key={timezone} className="timezone">
                    {timezone}
                  </span>
                ))}
              </span>
            </li>
            <li>
              <span id="clock" />
              {' '}
              {/* added element with ID */}
            </li>
          </ul>
          <div />
        </div>
      )}
    </>
  );
};

export default CountryInfo;
