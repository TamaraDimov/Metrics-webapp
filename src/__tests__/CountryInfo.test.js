import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CountryInfo from '../components/CountryInfo';

describe('Details', () => {
  it('should render the component', () => {
    const detail = renderer
      .create(
        <Provider store={store}>
          <CountryInfo />
        </Provider>,
      )
      .toJSON();
    expect(detail).toMatchSnapshot();
  });
});
