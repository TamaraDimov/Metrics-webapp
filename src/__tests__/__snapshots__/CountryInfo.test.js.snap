// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryInfo renders component correctly 1`] = `<div />`;

exports[`Details should render the component 1`] = `null`;
